### PR TITLE
CORE-13858 Keep session manager metrics as a member variables

### PR DESCRIPTION
**Pull request by williamvigorr3**
This prevents them being garbage collected. If they are garbage collected it means micrometer can't read them.

### CODE REVIEW

**Review by charlieR3 — DISMISSED** [review]
LGTM

File: components/link-manager/src/main/kotlin/net/corda/p2p/linkmanager/sessions/SessionManagerImpl.kt [hunk 1]
```diff
@@ -901,11 +901,12 @@ internal class SessionManagerImpl(
         return false
     }
 
+    private val outboundSessionCountMetric = CordaMetrics.Metric.OutboundSessionCount.builder().build()
+    private val inboundSessionContMetric = CordaMetrics.Metric.InboundSessionCount.builder().build()
+
```
Comment (dimosr) [inline components/link-manager/src/main/kotlin/net/corda/p2p/linkmanager/sessions/SessionManagerImpl.kt]: very minor but can we move these along with the other member variables at the top?

Comment (charlieR3) [reply components/link-manager/src/main/kotlin/net/corda/p2p/linkmanager/sessions/SessionManagerImpl.kt]: Just noticed a typo here also `inboundSessionContMetric` -> `inboundSessionCountMetric`

Comment (williamvigorr3) [reply components/link-manager/src/main/kotlin/net/corda/p2p/linkmanager/sessions/SessionManagerImpl.kt]: I fixed these

**Review by charlieR3 — APPROVED** [review]

**Review by dimosr — APPROVED** [review]

**Review by driessamyn — APPROVED** [review]
LGTM

**Pull request merged**